File: decompiled/3.17.1/be/ehealth/businessconnector/wsconsent/exception/WsConsentBusinessConnectorException.java
package be.ehealth.businessconnector.wsconsent.exception;

import be.ehealth.technicalconnector.exception.ConnectorException;
import java.text.MessageFormat;

public class WsConsentBusinessConnectorException extends ConnectorException {
   private static final long serialVersionUID = 1L;

   public WsConsentBusinessConnectorException(WsConsentBusinessConnectorExceptionValues errorCodeValue, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode());
   }

   public WsConsentBusinessConnectorException(WsConsentBusinessConnectorExceptionValues errorCodeValue, Throwable e, Object... params) {
      super(MessageFormat.format(errorCodeValue.getMessage(), params), errorCodeValue.getErrorCode(), e);
   }
}

File: src/main/kotlin/org/taktik/connector/business/wsconsent/exception/WsConsentBusinessConnectorException.kt
package org.taktik.connector.business.wsconsent.exception

import org.taktik.connector.technical.exception.ConnectorException
import java.text.MessageFormat

class WsConsentBusinessConnectorException : ConnectorException {

    constructor(
        errorCodeValue: WsConsentBusinessConnectorExceptionValues,
        vararg params: Any
    ) : super(MessageFormat.format(errorCodeValue.message, *params), errorCodeValue.errorCode) {
    }

    constructor(errorCodeValue: WsConsentBusinessConnectorExceptionValues, e: Throwable, vararg params: Any) : super(
        MessageFormat.format(errorCodeValue.message, *params),
        errorCodeValue.errorCode,
        e
    ) {
    }

    companion object {
        private val serialVersionUID = 1L
    }
}
